### PR TITLE
Build with Gradle

### PR DESCRIPTION
**Pull request by reta**
### Description
Build with Gradle

### Issues Resolved
Depends on https://github.com/opensearch-project/spring-data-opensearch/pull/35
Closes https://github.com/opensearch-project/spring-data-opensearch/issues/17

### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [X] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: src/main/java/org/opensearch/data/client/orhlc/ClientConfigurationBuilder.java [hunk 1]
```diff
@@ -9,7 +9,6 @@
 
 package org.opensearch.data.client.orhlc;
 
-
```
Comment (reta) [inline src/main/java/org/opensearch/data/client/orhlc/ClientConfigurationBuilder.java:12]: For some reasons Gradle's `spotless` does not like this new line, so had to reformat a bit ...

**Review by dblock — APPROVED** [review]
LGTM!

**Comment by dblock** [comment]
Rebase pls (I think a dependabot PR got merged). Also do you want testcontainers to go firsts?

**Review by rursprung — COMMENTED** [review]
just saw this now - thanks a lot for this! some small questions, nothing important

File: build.gradle.kts [hunk 2]
```diff
@@ -0,0 +1,227 @@
+/*
```
[the remark this answers is not in the document]
Comment (reta) [reply build.gradle.kts:1]: Thanks @rursprung , yes, supporting Gradle builds is a nigthmare, with Koltin it is a bit easier due to static nature of the language (but I am not fun of either Groovy or Kotlin). A number of OpenSearch projects use Kotlin DSL, fe https://github.com/opensearch-project/opensearch-java

**Comment by dlvenable** [comment]
Nice! Thanks @reta !

**Pull request merged**